Rationale.md

```diff
@@ -13,10 +13,17 @@ codebases, we'll revisit the alternatives listed below, reevaluate the tradeoffs
 and update the [design](AstSemantics.md) before the MVP is finalized.
 
 
-## Why AST?
+## Why a stack machine?
 
-Why not a register- or SSA-based bytecode?
-* Trees allow a smaller binary encoding: [JSZap][], [Slim Binaries][].
+Why not an AST, or a register- or SSA-based bytecode?
+
+* We started with an AST and generalized to a [structured stack machine](AstSemantics.md). ASTs allow a 
+  dense encoding and efficient decoding, compilation, and interpretation.
+  The stack machine is a generalization of ASTs allowed in previous versions while allowing
+  efficiency gains in interpretation and baseline compilation, as well as a straightforward 
+  design for multi-return functions.
+* The stack machine allows smaller binary encoding than registers or SSA, and structured control
+  flow preserves the size advantages of an AST: [JSZap][], [Slim Binaries][].
```

Comment:
I can imagine encodings of basic-blocks+goto that would be roughly the same size as what we have with blocks and branches.  I think the better argument is that structured control flow simplifies validation, SSA construction, and various other compiler-backend algorithms.